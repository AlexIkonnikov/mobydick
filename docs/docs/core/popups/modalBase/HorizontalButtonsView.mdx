import HorizontalButtonsView from '../assets/HorizontalButtonsView.png';

<img src={HorizontalButtonsView} width="50%"/>

Reusable HorizontalButtonsView component for ModalBase

## Example

```tsx
import {
  Button,
  createStyles,
  IButtonTypes,
  ModalBase,
  Typography,
  usePopups,
  useStyles,
  View,
} from '@lad-tech/mobydick-core';

export const ModalWidget = () => {
  const [styles] = useStyles(styleFn);
  const {openPopup} = usePopups();

  return (
    <View style={styles.container}>
      <Typography font={'Regular-Primary-H5'}>Modal</Typography>
      <Button
        text={'ModalBase'}
        onPress={() =>
          openPopup({
            Content: props => (
              <ModalBase {...props}>
                <ModalBase.HorizontalButtonsView
                  textLeft={'Left btn'}
                  textRight={'Right btn'}
                  typeLeft={IButtonTypes.primary}
                  typeRight={IButtonTypes.destructive}
                  onPressRight={props.onClose}
                  onPressLeft={props.onClose}
                />
              </ModalBase>
            ),
          })
        }
      />
    </View>
  );
};

const styleFn = createStyles(({spaces}) => ({
  container: {
    gap: spaces.Space8,
  },
}));
```


## `Props`

### <div class="label required basic">Required</div>**`typeLeft`**

| TYPE                                                                                                                |
|:--------------------------------------------------------------------------------------------------------------------|
| [IButtonTypes](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/cta/components/Button/types.ts#L20) |

TypeLeft for left button in HorizontalButtonsView


### <div class="label required basic">Required</div>**`typeRight`**

| TYPE                                                                                                                |
|:--------------------------------------------------------------------------------------------------------------------|
| [IButtonTypes](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/cta/components/Button/types.ts#L20) |

TypeRight for right button in HorizontalButtonsView

### <div class="label required basic">Required</div>**`onPressLeft`**

| TYPE       |
|:-----------|
| () => void |

OnPress for left button in HorizontalButtonsView

### <div class="label required basic">Required</div>**`onPressRight`**

| TYPE       |
|:-----------|
| () => void |

OnPress for right button in HorizontalButtonsView

### <div class="label required basic">Required</div>**`textLeft`**

| TYPE   |
|:-------|
| string |

Text for left button in HorizontalButtonsView

### <div class="label required basic">Required</div>**`textRight`**

| TYPE   |
|:-------|
| string |

Text for right button in HorizontalButtonsView

### `disabledLeft`

| TYPE    |
|:--------|
| boolean |

Disabled for left button in HorizontalButtonsView

### `disabledRight`

| TYPE    |
|:--------|
| boolean |

Disabled for right button in HorizontalButtonsView
