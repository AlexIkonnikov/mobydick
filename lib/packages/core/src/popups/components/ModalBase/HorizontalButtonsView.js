"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const useStyles_1 = __importDefault(require("../../../styles/hooks/useStyles"));
const View_1 = __importDefault(require("../../../basic/components/View/View"));
const Button_1 = __importDefault(require("../../../cta/components/Button/Button"));
const types_1 = require("../../../cta/components/Button/types");
const stylesCreate_1 = __importDefault(require("./stylesCreate"));
const HorizontalButtonsView = props => {
    const [styles] = (0, useStyles_1.default)(stylesCreate_1.default);
    const { typeLeft, onPressLeft, textLeft, typeRight, textRight, onPressRight, disabledRight, disabledLeft, } = props;
    return (<View_1.default style={styles.horizontalButtonsView}>
      <Button_1.default size={types_1.IButtonSize.fixed} style={styles.horizontalLeftButton} type={typeLeft} onPress={onPressLeft} text={textLeft} disabled={disabledLeft}/>
      <Button_1.default size={types_1.IButtonSize.fixed} style={styles.horizontalRightButton} type={typeRight} onPress={onPressRight} text={textRight} disabled={disabledRight}/>
    </View_1.default>);
};
exports.default = HorizontalButtonsView;
