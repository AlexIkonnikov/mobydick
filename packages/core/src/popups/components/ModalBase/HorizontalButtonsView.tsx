import {FC} from 'react';

import {IHorizontalButtonsView} from '../../types';
import useStyles from '../../../styles/hooks/useStyles';
import View from '../../../basic/components/View/View';
import Button from '../../../cta/components/Button/Button';
import {IButtonSize} from '../../../cta/components/Button/types';

import stylesCreate from './stylesCreate';

const HorizontalButtonsView: FC<IHorizontalButtonsView> = props => {
  const [styles] = useStyles(stylesCreate);
  const {
    typeLeft,
    onPressLeft,
    textLeft,
    typeRight,
    textRight,
    onPressRight,
    disabledRight,
    disabledLeft,
  } = props;

  return (
    <View style={styles.horizontalButtonsView}>
      <Button
        size={IButtonSize.fixed}
        style={styles.horizontalLeftButton}
        type={typeLeft}
        onPress={onPressLeft}
        text={textLeft}
        disabled={disabledLeft}
      />
      <Button
        size={IButtonSize.fixed}
        style={styles.horizontalRightButton}
        type={typeRight}
        onPress={onPressRight}
        text={textRight}
        disabled={disabledRight}
      />
    </View>
  );
};

export default HorizontalButtonsView;
