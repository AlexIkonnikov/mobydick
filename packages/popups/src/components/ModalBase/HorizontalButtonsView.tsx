import React, {FC} from 'react';
import {Button, IButtonSize, IButtonTypes} from '@npm/mobydick-cta';
import {View} from '@npm/mobydick-core';
import {useStyles} from '@npm/mobydick-styles';

import stylesCreate from './stylesCreate';

interface IFilledView {
  typeLeft: IButtonTypes;
  onPressLeft(): void;
  textLeft: string;

  typeRight: IButtonTypes;
  onPressRight(): void;
  textRight: string;
}

const HorizontalButtonsView: FC<IFilledView> = props => {
  const [styles] = useStyles(stylesCreate);
  const {typeLeft, onPressLeft, textLeft, typeRight, textRight, onPressRight} =
    props;

  return (
    <View style={styles.horizontalButtonsView}>
      <Button
        size={IButtonSize.fixed}
        style={styles.horizontalLeftButton}
        type={typeLeft}
        onPress={onPressLeft}
        text={textLeft}
      />
      <Button
        size={IButtonSize.fixed}
        style={styles.horizontalRightButton}
        type={typeRight}
        onPress={onPressRight}
        text={textRight}
      />
    </View>
  );
};

export default HorizontalButtonsView;
