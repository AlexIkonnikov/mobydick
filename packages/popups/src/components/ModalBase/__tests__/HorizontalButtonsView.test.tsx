import {render} from '@testing-library/react-native';
import React from 'react';
import {ITypes} from '@npm/mobydick-cta';

import HorizontalButtonsView from '../HorizontalButtonsView';

describe('@npm/mobydick-popups/modalBase', () => {
  it('should renders correctly HorizontalButtonsView destructive ', () => {
    const {toJSON} = render(
      <HorizontalButtonsView
        typeLeft={ITypes.destructive}
        onPressLeft={() => console.log('onPressLeft')}
        textLeft={'textLeft'}
        typeRight={ITypes.destructive}
        onPressRight={() => console.log('onPressRight')}
        textRight={'textRight'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly HorizontalButtonsView secondary ', () => {
    const {toJSON} = render(
      <HorizontalButtonsView
        typeLeft={ITypes.secondary}
        onPressLeft={() => console.log('onPressLeft')}
        textLeft={'textLeft'}
        typeRight={ITypes.secondary}
        onPressRight={() => console.log('onPressRight')}
        textRight={'textRight'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly HorizontalButtonsView tertiary ', () => {
    const {toJSON} = render(
      <HorizontalButtonsView
        typeLeft={ITypes.tertiary}
        onPressLeft={() => console.log('onPressLeft')}
        textLeft={'textLeft'}
        typeRight={ITypes.tertiary}
        onPressRight={() => console.log('onPressRight')}
        textRight={'textRight'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly HorizontalButtonsView primary ', () => {
    const {toJSON} = render(
      <HorizontalButtonsView
        typeLeft={ITypes.primary}
        onPressLeft={() => console.log('onPressLeft')}
        textLeft={'textLeft'}
        typeRight={ITypes.primary}
        onPressRight={() => console.log('onPressRight')}
        textRight={'textRight'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
