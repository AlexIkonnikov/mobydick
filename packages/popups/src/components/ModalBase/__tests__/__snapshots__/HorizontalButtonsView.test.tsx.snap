// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/modalBase should renders correctly HorizontalButtonsView destructive  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 36,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#F54D3D",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#FFF",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textLeft
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#F54D3D",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#FFF",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textRight
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-popups/modalBase should renders correctly HorizontalButtonsView primary  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 36,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#2B78EE",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#FFF",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textLeft
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#2B78EE",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#FFF",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textRight
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-popups/modalBase should renders correctly HorizontalButtonsView secondary  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 36,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "rgba(43, 120, 238, 0.15)",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#2B78EE",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textLeft
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "rgba(43, 120, 238, 0.15)",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#2B78EE",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textRight
    </Text>
  </View>
</View>
`;

exports[`@npm/mobydick-popups/modalBase should renders correctly HorizontalButtonsView tertiary  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 36,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginRight": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#2B78EE",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textLeft
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderRadius": 18,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 9,
        "minHeight": 72,
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={true}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 27,
            "lineHeight": 33,
          },
          Array [
            Object {
              "color": "#2B78EE",
              "paddingHorizontal": 9,
            },
            undefined,
          ],
        ]
      }
    >
      textRight
    </Text>
  </View>
</View>
`;
